src/mlpack/methods/ann/layer/noisylinear_impl.hpp

```diff
@@ -48,6 +48,50 @@ NoisyLinear<InputDataType, OutputDataType>::NoisyLinear(
   biasEpsilon.set_size(outSize, 1);
 }
 
+template<typename InputDataType, typename OutputDataType>
+NoisyLinear<InputDataType, OutputDataType>::NoisyLinear(
+    NoisyLinear&& layer) :
+    inSize(std::move(layer.inSize)),
+    outSize(std::move(layer.outSize)),
+    weights(std::move(layer.weights))
+{
+  layer.inSize = 0;
+  layer.outSize = 0;
+  layer.weights = nullptr;
```

Comment:
Is this the right thing to do here? `layer.weights` should have type `OutputDataType`, which will typically be `arma::mat` or some other matrix type---not a pointer. So I think the `std::move()` call in the initialization list should be sufficient.

Reply:
Same comment, I'll make change to this. Thanks!